src/Hypergrid.js

```diff
@@ -2190,6 +2205,15 @@ function setStyles(el, style, keys) {
     }
 }
 
+function stringifyFunctions() {
+    var self = this;
+    return Object.keys(this).reduce(function(obj, key) {
```

Comment:
As explained above, this design encompasses a map. It uses a map if provided; and constructs a map if not provided. The next saveState will then persist a map.